File: item-list/src/test/java/com/gurunars/item_list/FetcherComplexPermutationTest.java
package com.gurunars.item_list;

import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class FetcherComplexPermutationTest {

    private FetcherComplexPermutation<AnimalItem> fetcher = new FetcherComplexPermutation<>();

    @Test
    public void getNormal() {
        assertEquals(fetcher.apply(
                Arrays.asList(
                        new AnimalItem(1, 1),
                        new AnimalItem(2, 1),
                        new AnimalItem(3, 1),
                        new AnimalItem(4, 1)
                ),
                Arrays.asList(
                        new AnimalItem(1, 1),
                        new AnimalItem(3, 1),
                        new AnimalItem(2, 1),
                        new AnimalItem(4, 1)
                )),
                new ChangeComplexPermutation<>(1, Arrays.asList(
                        new AnimalItem(3, 1),
                        new AnimalItem(2, 1)
                ))
        );
    }
}

File: item-list/src/test/java/com/gurunars/item_list/FetcherComplexPermutationTest.kt
package com.gurunars.item_list

import org.junit.Test

import java.util.Arrays

import org.junit.Assert.assertEquals

class FetcherComplexPermutationTest {

    private val fetcher = FetcherComplexPermutation<AnimalItem>()

    @Test
    fun getNormal() {
        assertEquals(fetcher.apply(
                Arrays.asList(
                        AnimalItem(1, 1),
                        AnimalItem(2, 1),
                        AnimalItem(3, 1),
                        AnimalItem(4, 1)
                ),
                Arrays.asList(
                        AnimalItem(1, 1),
                        AnimalItem(3, 1),
                        AnimalItem(2, 1),
                        AnimalItem(4, 1)
                ), 0),
                ChangeComplexPermutation(1, Arrays.asList(
                        AnimalItem(3, 1),
                        AnimalItem(2, 1)
                ))
        )
    }
}